### ISSUE TITLE
E-mail not treated as case-insensitive during Admin UI login

### ISSUE THREAD

**Issue by JohnnyEstilles** [keystonejs/keystone#1323]
A Keystone consumer reported on [StackOverflow](http://stackoverflow.com/questions/29759761/i-cant-access-the-admin-ui-in-keystonejs) not being able to log in to the Admin UI with an e-mail address that had uppercase characters. (Thanks to @mikeLspohn for finding this bug.)

I found that `email.updateItem` coverts e-mail addresses to lowercase (see [fields/types/email/EmailType.js line 66](https://github.com/keystonejs/keystone/blob/master/fields/types/email/EmailType.js#L66)), but `session.login` does no such conversion prior to lookup (see [lib/session.js line 78](https://github.com/keystonejs/keystone/blob/master/lib/session.js#L78)).

Submitting PR to fix.

**Comment by creynders** [comment]
I wouldn't convert it to lower case since if the e-mail address isn't in lowercase in the DB, it won't match. Rather use a regex query in mongoose: 

```js
var email  = "info@Creynders.be";
User.model.findOne({email: new RegExp(email, "i")})
```

**Comment by JohnnyEstilles** [comment]
@creynders ... that is exactly what my PR does. Submitting in a few.

**Comment by creynders** [comment]
ok, just wanted to make sure :smile:

**Comment by JohnnyEstilles** [comment]
No worries. I know we all love Keystone and want to protect it. :-)

**Comment by JohnnyEstilles** [comment]
Closed via #1326 ...
